pkg/query/querier.go

```diff
@@ -331,6 +348,10 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 		warns: warns,
 	}
 
+	if allPushedDown && (strings.Contains(hints.Func, "max") || strings.Contains(hints.Func, "min")) {
```

Comment:
Note from the meeting: if the proxied data come here from different sources i.e. pushed down vs. not pushed down, we fall back to the old dedup iterator and in the results, you will see the same problem on time ranges where data comes from mixed sources. Ideally, we should have one iterator that handles this.